modules/RTC/RTCUtils.js

```diff
@@ -690,6 +711,48 @@ var RTCUtils = {
                     };
                 }
             }
+            else if (RTCBrowserType.isEdge()) {
+                var self = this;
+                var devices = navigator.mediaDevices;
+
+                this.getUserMedia = wrapGetUserMediaPromise(
+                    devices.getUserMedia.bind(devices)
+                );
+                this.attachMediaStream = wrapAttachMediaStream(function (element, stream) {
+                    if (!element) {
+                        return;
+                    }
+                    element.srcObject = stream;
+                    element.play();
+
+                    return element;
+                });
+                this.enumerateDevices = wrapEnumerateDevices(
+                    devices.enumerateDevices.bind(devices)
+                );
+                this.getStreamID = function (stream) {
+                    var id = stream.id;
+                    if (!id) {
+                        var tracks = stream.getVideoTracks();
+                        if (!tracks || tracks.length === 0) {
+                            tracks = stream.getAudioTracks();
+                        }
+                        id = tracks[0].id;
+                    }
+                    return SDPUtil.filter_special_chars(id);
+                };
```

Comment:
That getStreamID is a 100% copy of the Firefox case. It could be extracted into a function of its own to be reused by both Firefox and Edge.